### PR TITLE
Refactor structure of the bcdaworker

### PR DESCRIPTION
**Pull request by mtrang1263**
### Overview
To enable us to add more functionality to the bcdaworker as well as add the capability of alternative queueing libraries, we need to refactor the bcdaworker to separate the functionalities that currently exist in main.go

### Change Details
1. Introduce worker type that will be responsible for handling the job processing. This worker contains most of the business logic that was captured in bcdaworker's main.go
2. Introduce manager type that will be responsible for deserializing the job and calling the worker to perform the various tasks. This manager type is implemented with the logic needed to retrieve work from the queue. In our case, we have one implementation that uses the que-go client to deserialize a que.Job into the underlying JobEnqueueArgs. 

    This will allow us to introduce other manager impls that will be responsible for coordinating work.

3. Introduce a enqueue type that is responsible for serializing a models.JobEnqueueArgs into the queue data type. In our case, we have one implementation that uses que-go to add work to the queue.

    This will allow us to introduce other enqueue impls that will be responsible for translating JobEnqueueArgs to an underlying queue datatype.

### Security Implications

- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [x] no PHI/PII is affected by this change

### Acceptance Validation
1. Add tests to verify the behavior of the worker, manager, and enqueuer work as expected.
2. TODO - Create follow up JIRA tickets to integrate the enqueuer into the BCDA API. Right now it instantiates the que client directly.
3. Deployed to dev and validated [smoke tests](https://bcda-ci.adhocteam.us/job/BCDA%20-%20Test%20-%20Integration%20Smoke/2860/) pass.

### CODE REVIEW

File: bcdaworker/worker/worker_test.go [hunk 1]
```diff
@@ -425,7 +416,7 @@ func (s *MainTestSuite) TestProcessJobEOB() {
 	assert.Contains(s.T(), []models.JobStatus{models.JobStatusFailed, models.JobStatusCompleted}, completedJob.Status)
 }
 
-func (s *MainTestSuite) TestProcessJob_EmptyBasePath() {
```
Comment (mtrang1263) [inline bcdaworker/worker/worker_test.go:428]: Test moved into `TestValidateJob`

File: bcdaworker/worker/worker_test.go [hunk 2]
```diff
@@ -444,39 +435,28 @@ func (s *MainTestSuite) TestProcessJob_EmptyBasePath() {
...
 	assert.EqualError(s.T(), err, "empty BBBasePath: Must be set")
-}
+	assert.Nil(s.T(), j1)
 
-func (s *MainTestSuite) TestProcessJob_InvalidArgs() {
-	j := que.Job{Args: []byte("{ this is not valid JSON }")}
-	assert.EqualError(s.T(), processJob(&j), "invalid character 't' looking for beginning of object key string")
 }
 
-func (s *MainTestSuite) TestProcessJob_InvalidJobID() {
```
Comment (mtrang1263) [inline bcdaworker/worker/worker_test.go:462]: Test moved into `TestValidateJob`

File: bcdaworker/worker/worker_test.go [hunk 3]
```diff
@@ -486,94 +466,22 @@ func (s *MainTestSuite) TestProcessJob_NoBBClient() {
...
+	assert.Contains(s.T(), s.w.ProcessJob(context.Background(), j, jobArgs).Error(), "could not create Blue Button client")
 }
 
-func (s *MainTestSuite) TestSetupQueue() {
-	setupQueue()
-	os.Setenv("WORKER_POOL_SIZE", "7")
-	setupQueue()
-}
-
-func (s *MainTestSuite) TestUpdateJobStats() {
```
Comment (mtrang1263) [inline bcdaworker/worker/worker_test.go:515]: Check added `TestProcessJobEOB` to verify we update the CompletedJobCount

**Review by adamayd — APPROVED** [review]
LGTM!!  Kudos on the amount of work to do this.

File: bcdaworker/worker/worker.go [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+	"github.com/CMSgov/bcda-app/bcda/models"
+	fhirmodels "github.com/CMSgov/bcda-app/bcda/models/fhir"
+	"github.com/CMSgov/bcda-app/bcda/responseutils"
+	"github.com/CMSgov/bcda-app/bcda/utils"
+	"github.com/CMSgov/bcda-app/bcdaworker/repository"
+	"github.com/CMSgov/bcda-app/bcdaworker/repository/postgres"
+	"github.com/CMSgov/bcda-app/conf"
+	newrelic "github.com/newrelic/go-agent"
+	"github.com/pborman/uuid"
+	"github.com/pkg/errors"
```
Comment (adamayd) [inline bcdaworker/worker/worker.go:24]: Now that I'm working through this a bit, could with alias this package to something like `pkgerrors` and leave the go std `errors`?

Comment (mtrang1263) [reply bcdaworker/worker/worker.go:24]: ah that's a great idea

Comment (mtrang1263) [reply bcdaworker/worker/worker.go:24]: Will create a ticket to address this in all packages.

**Pull request merged**